### DISCUSSION TITLE
switch wallet account directly

### DISCUSSION THREAD

**Discussion by PaulNext** [rainbow-me/rainbowkit#1012] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/100060836/217840958-a447ef44-ea39-4c77-8478-e98adde155e6.png)
![image](https://user-images.githubusercontent.com/100060836/217841055-ff3b1b03-6a78-47e6-b1a1-4019e0bd07fb.png)
Hi buddy, our products have been using RainbowKit for a while. One of the interaction scenarios is to directly switch the user's wallet and directly invoke the wallet plug-in of metamask or coinbase. I found some documents and interfaces in the community but did not find this api. I hope the official website The staff will evaluate whether this hooks API can be opened to our community developers.

**Comment by crsxmd** [comment]
any update for this ?

**Comment by DanielSinclair** [comment, marked as answer]
Unfortunately it is not currently possible to switch accounts directly from Wagmi or RainbowKit, as this is handled purely within the wallet (i.e. MetaMask). Wagmi relies on communication via RPC with the MetaMask Provider, and this particular call unfortunately doesn't exist yet. This is something that Rainbow will address in the coming months in our own product suite.
